library/core/src/intrinsics/mod.rs

```diff
@@ -2629,6 +2629,7 @@ pub const fn overflow_checks() -> bool {
 #[rustc_nounwind]
 #[rustc_intrinsic]
 #[miri::intrinsic_fallback_is_spec]
+#[rustc_intrinsic_const_stable_indirect]
 pub const unsafe fn const_allocate(_size: usize, _align: usize) -> *mut u8 {
```

Comment:
this and make_global are needed in https://github.com/rust-lang/rust/blob/e2893f7c95e45f74c8bc5dbd033486c5bfaa4deb/library/alloc/src/boxed/thin.rs#L334

Reply:
This one *really* shouldn't be callable-from-stable, it is the very definition of not stable... same for `const_make_global`.

Reply:
I would strongly prefer to add allow_const_unstable to `new_unsize_zst` -- and we should consider if it's really safe to use these highly experimental features there.

That said, `Thin` is entirely unstable, so we can also add `allow_const_unstable` with a comment saying that this isn't actually stable -- but then we should also add that as a blocker on the tracking issue as it seems easy to miss.

Reply:
> I would strongly prefer to add allow_const_unstable to new_unsize_zst
That does feel nicer, to guard it against any other uses. With a comment there why specifically this call needs it and is safe